reseq_ros2/launch/reseq_core_launch.py

```diff
@@ -0,0 +1,152 @@
+                'r_head_roll_vel': config['scaler_consts']['r_head_roll_vel'],
+            }]))
+
+    if config['version'] == 'mk1':
+        launch_config.append(Node(
+                package='reseq_ros2',
+                executable='enea',
+                name='enea',
+                parameters=[{
+                    'pitch': config['enea_consts']['i_pitch'],
+                    'head_pitch': config['enea_consts']['i_head_pitch'],
+                    'head_roll': config['enea_consts']['i_head_roll'],
+                    'servo_speed': config['enea_consts']['servo_speed'],
+                    'r_pitch': config['enea_consts']['r_pitch'],
+                    'r_head_pitch': config['enea_consts']['r_head_pitch'],
+                    'r_head_roll': config['enea_consts']['r_head_roll'],
+                    'pitch_conv': config['enea_consts']['pitch_conv'],
+                    'end_effector': endEffector
+                }]))
+    robot_controllers = f"{config_path}/reseq_controllers.yaml"
+    control_node = Node(
+        package="controller_manager",
+        executable="ros2_control_node",
+        parameters=[robot_controllers],
+        output="both",
+        remappings=[
+            ("~/robot_description", "/robot_description"),
+        ],
+    )
+    launch_config.append(control_node)
+
+    joint_state_broadcaster_spawner = Node(
+        package="controller_manager",
+        executable="spawner",
+        arguments=["joint_state_broadcaster", "--controller-manager", "/controller_manager"],
+    )
+    launch_config.append(joint_state_broadcaster_spawner)
+
+    diff_controller_spawner1 = Node(
+        package="controller_manager",
+        executable="spawner",
+        arguments=["diff_controller1", "--controller-manager", "/controller_manager"],
+    )
+    launch_config.append(diff_controller_spawner1)
+
+    diff_controller_spawner2 = Node(
+        package="controller_manager",
+        executable="spawner",
+        arguments=["diff_controller2", "--controller-manager", "/controller_manager"],
+    )
+    launch_config.append(diff_controller_spawner2)
+
+    # frf = Node(
+    #     package='reseq_ros2',
+    #     executable='fake_robot_feedback',
+    #     name='fake_robot_feedback',
+    #     parameters=[{
+    #         'modules': addresses,
+    #     }]
+    # )
+    # launch_config.append(frf)
+
+    # all other reseq_core nodes are core nodes
+    core_nodes += list(range(1, len(launch_config)))
+
+    event_handlers = []
+    # create list of event handlers for each core node
+    for el in core_nodes:
+        event_handlers.append(
+            RegisterEventHandler(
+                event_handler=OnProcessExit(
+                    target_action = launch_config[el],
+                    on_exit=[
+                        LogInfo(msg=f'{launch_config[el].name} exited, system ends'),
+                        EmitEvent(event=Shutdown())
+                    ]
+                )
+            )
+        )
```

Comment:
This is an interesting approach, however as we suggested in the meeting, this approach does not consider the special case of the communication node (which can die if we are in the VCAN configuration but is required in the CAN configuration). 

Moreover I believe that we could just add for each node the event directly as shown [here](https://github.com/MetroRobots/rosetta_launch?tab=readme-ov-file#10---make-a-node-required) and use an if switch for deciding the communication node

Reply:
The case where channel is "VCAN", the core_nodes won't contain the index of the communication node, so it won't have the event handler, this case is handled by these lines:
```python
    if config['canbus']['channel'].startswith("can"):
        core_nodes.append(0)
```
An alternative solution is to declare the node as required with on_exit=Shutdown(), but the above-mentioned solution adds more flexibility. If you still think that the easier solution is enough let me know